Helpers/TimeTableActionHandler.php

```diff
@@ -0,0 +1,206 @@
+        $timesheetId = isset($postData['timesheet-id']) ? (int)$postData['timesheet-id'] : 0;
+        $workDate = CarbonImmutable::createFromFormat('Y-m-d', $postData['timesheet-date'])->startOfDay();
+        $workDate = $workDate->setToDbTimezone();
+
+        $values = [
+            'timesheetId' => $postData['timesheet-id'],
+            'userId' => session('userdata.id'),
+            'hours' => $postData['timesheet-hours'],
+            'workDate' => $workDate,
+            'ticketId' => $postData['timesheet-ticket-id'],
+            'description' => $postData['timesheet-description'],
+            'kind' => 'GENERAL_BILLABLE',
+        ];
+        $this->timeTableService->updateOrAddTimelogOnTicket($values, $timesheetId);
+
+        // Delegate query parameter addition to appendQueryParams
+        return $this->appendQueryParams($postData, $redirectUrl);
+    }
+
+    /**
+     * Deletes a ticket based on the provided POST data and redirects to the specified URL.
+     * Outputs a JSON-encoded response indicating success or failure status.
+     *
+     * @param array<string, mixed> $postData    Postdata
+     * @param string               $redirectUrl Redirect url.
+     * @return void Json encoded return.
+     */
+    public function deleteTicket(array $postData, string $redirectUrl): void
+    {
+        $timesheetId = $postData['timesheetId'];
+        $redirectUrl = $this->appendQueryParams($postData, $redirectUrl);
+
+        if ($timesheetId) {
+            try {
+                $this->timesheetRepository->deleteTime($timesheetId);
+                exit(json_encode(['status' => 'success', 'redirectUrl' => $redirectUrl]));
+            } catch (\Exception $e) {
+                exit(json_encode(['status' => 'error', 'error' => $e->getMessage()]));
+            }
+        } else {
+            exit(json_encode(['status' => 'error', 'error' => 'Missing timesheetId in POST data.']));
+        }
+    }
+
+    /**
+     * Appends appropriate query parameters based on POST data to the given redirect URL.
+     *
+     * @param array<string, mixed> $postData    The POST data for query parameters.
+     * @param string               $redirectUrl The base URL to which query parameters will be appended.
+     * @return string The updated redirect URL with appended query parameters.
+     */
+    private function appendQueryParams(array $postData, string $redirectUrl): string
+    {
+        $queryParams = [];
+
+        // Use fromDate and toDate from POST if set, otherwise check GET
+        $queryParams['fromDate'] = $postData['fromDate'] ?? $_GET['fromDate'] ?? null;
+        $queryParams['toDate'] = $postData['toDate'] ?? $_GET['toDate'] ?? null;
+
+        // Remove null values
+        $queryParams = array_filter($queryParams);
+
+        // Add query parameters to the URL if necessary
+        if (!empty($queryParams)) {
+            $redirectUrl .= (!str_contains($redirectUrl, '?') ? '?' : '&') . http_build_query($queryParams);
```

Comment:
Maybe https://www.php.net/manual/en/function.parse-url.php with the http_build_query could make this a little more readable and secure?